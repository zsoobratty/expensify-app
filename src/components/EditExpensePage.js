import React from 'react'

const EditExpensePage = (props) => {
    console.log(props)
    return (
        <div>
            Editing the expense with the ID of {props.match.params.id}
        </div>
    )
}

export default EditExpensePage