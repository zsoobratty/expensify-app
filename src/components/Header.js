import React from 'react'
import { NavLink } from 'react-router-dom'

const Header = () => (
    <header>
        <h1>Expensify</h1>
        <NavLink exact to='/' activeClassName='is-active'>Dashboard</NavLink>
        <NavLink to='/create' activeClassName='is-active'>Create Expenses</NavLink>
        <NavLink to='/help' activeClassName='is-active'>Help</NavLink>
    </header>
)

export default Header