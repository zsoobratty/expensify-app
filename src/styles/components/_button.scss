// Big button

.big-button {
    background-color: $dark-purple;
    border: none;
    border-bottom: .6rem solid darken($dark-purple, 50%);
    border-left: .6rem solid darken($dark-purple, 25%);
    
    color: $light-purple;
    font-weight: bold;
    font-size: $l-size;
    margin-bottom: $m-size;
    padding: 2.4rem;
    width: 100%;
}

.big-button:disabled {
    opacity: .5;
}


@media(min-width: $desktop-breakpoint) {
    .big-button {
        margin-bottom: $xl-size;
    }
}

// Button

.button {
    background: $dark-purple;
    border: none;
    border-bottom: .3rem solid darken($dark-purple, 10%);
    color: $light-purple;
    font-weight: 500;
    padding: $s-size
}

.button--link {
    background: none;
    border: none;
    color: $off-white;
    padding: 0;
}
