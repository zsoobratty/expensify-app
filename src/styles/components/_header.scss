.header {
    background: $dark-blue;
    color: white;
    margin-bottom: $m-size;
    padding: $m-size 0;
}

.header__title {
    font-size: $l-size;
    margin: 0;
}

.header_subtitle {
    color: $light-purple;
    font-size: $m-size;
    font-weight: 500;
    margin: 0;
}

@media (min-width: $desktop-breakpoint) {
    .header {
        margin-bottom: $xl-size;
    }
}