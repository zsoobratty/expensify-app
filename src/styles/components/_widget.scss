// Widget
.widget {
    background: $light-blue;
    margin-bottom: $xl-size;
}

.widget__message {
    color: $off-white;
    margin: 0;
    padding: $l-size;
    text-align: center;
    border-bottom: 0.3rem solid lighten($light-blue, 10%);
}

// Widget Header
.widget-header {
    background-color: $darkest-blue;
    color: $light-purple;
    display: flex;
    justify-content: space-between;
    padding: $m-size;

}

.widget-header__title {
    margin: 0;
}
