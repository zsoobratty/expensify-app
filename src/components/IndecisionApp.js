import React from 'react'
import AddOption from './AddOption'
import Action from './Action'
import Header from './Header'
import Options from './Options'
import OptionModal from'./OptionModal'

export default class IndecisionApp extends React.Component {
    state = {
        options: [],
        selectedOption: undefined
    }

    handleClearSelectedOption = () => {
        this.setState(() => ({ selectedOption: undefined }))
    }

    handleDeleteOptions = () => {
        this.setState(() => ({options: [] }))
    }

    handleDeleteOption = (optionToRemove) => {
        this.setState((prevState) => ({
            options: prevState.options.filter((option) => option !== optionToRemove )
        }))
    }

    handlePick = () => {
        const randomNum = Math.floor(Math.random() * this.state.options.length)
        const option = this.state.options[randomNum]
        this.setState(() => ({
            selectedOption: option
        }))
    }

    handleAddOption = (option) => {
        if (!option) {
            return 'Enter an item'
        } else if (this.state.options.indexOf(option) > -1) {
            return 'This option already exists'
        }
       
        this.setState((prevState) => ({ options: prevState.options.concat(option) }))
    }

    componentDidMount() {
        try {
            const json = localStorage.getItem('options')
            const options = JSON.parse(json)
            
            if (options) {
                this.setState(() => ({ options }))
            }
        } catch (e) {
            // Do nothing
        }
    }

    componentDidUpdate(prevProps, prevState) {
        if (prevState.options.length !== this.state.options.length) {
            const json = JSON.stringify(this.state.options)
            localStorage.setItem('options', json)
        }
    }

    componentWillUnmount() {
        console.log('componentWillUnmount')
    }

    render() {
        const subtitle = "We'll take it from here!"

        return(
            <div>
                <Header subtitle={subtitle} />
                <Action 
                    hasOptions={this.state.options.length > 0}
                    handlePick={this.handlePick}
                />
                <Options 
                    options={this.state.options} 
                    handleDeleteOptions={this.handleDeleteOptions}
                    handleDeleteOption={this.handleDeleteOption}
                />
                <AddOption 
                    handleAddOption={this.handleAddOption}
                />
                <OptionModal 
                    selectedOption={this.state.selectedOption}
                    handleClearSelectedOption={this.handleClearSelectedOption}
                />
            </div>
        )
    }
}
