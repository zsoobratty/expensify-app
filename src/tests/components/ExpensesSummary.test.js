import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={150} />)
    expect(wrapper).toMatchSnapshot();
})

test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={26} expensesTotal={12349124300} />)
    expect(wrapper).toMatchSnapshot();
})
