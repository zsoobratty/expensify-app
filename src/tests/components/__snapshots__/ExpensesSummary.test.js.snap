// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling £
    1.50
  </h1>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    26
     
    expenses
     totalling £
    123,491,243.00
  </h1>
</div>
`;
